src/hooks/useSovereignVenue.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from "react";
+
+import {
+  setSovereignVenue,
+  setSovereignVenueIsLoading,
+  setSovereignVenueError,
+} from "store/actions/SovereignVenue";
+
+import { fetchSovereignVenue } from "api/venue";
+
+import { AnyVenue } from "types/venues";
+import { ReactHook } from "types/utility";
+
+import { WithId } from "utils/id";
+import { sovereignVenueSelector } from "utils/selectors";
+import { tracePromise } from "utils/performance";
+
+import { useDispatch } from "./useDispatch";
+import { useSelector } from "./useSelector";
+
+export interface UseSovereignVenueProps {
+  venueId?: string;
+}
+
+export interface UseSovereignVenueData {
+  sovereignVenue?: WithId<AnyVenue>;
+  sovereignVenueId?: string;
+  isSovereignVenueLoading: boolean;
+  errorMsg?: string;
+}
+
+export const useSovereignVenue: ReactHook<
+  UseSovereignVenueProps,
+  UseSovereignVenueData
+> = ({ venueId }) => {
+  const dispatch = useDispatch();
+
+  const {
+    venue: sovereignVenue,
+    isLoading: isSovereignVenueLoading,
+    errorMsg,
+  } = useSelector(sovereignVenueSelector);
+
+  useEffect(() => {
+    // NOTE: Force to fetch it only once
```

Comment:
OOS, but this is something I don’t really like.. and we should really find a better solution for.

we want to only fetch it once for a ‘known venueId’ or any related venues. But if we ‘step outside’ that set of related venues, we absolutely need to find the new one.

This will cause problems when we switch to using only router based navigation and need to step across the boundary of related venues, which we already do in env/github

Reply:
@0xdevalias fixed in 98b0388

Reply:
What was fixed in the commit? @dmitrii-zolotukhin